oauthenticator/github.py

```diff
@@ -205,23 +226,32 @@ async def update_auth_model(self, auth_model):
         # - about /user/emails: https://docs.github.com/en/rest/reference/users#list-email-addresses-for-the-authenticated-user
         #
         # Note that the read:user scope does not imply the user:emails scope!
+        # Note that the retrieved scopes will be empty for GitHub Apps https://docs.github.com/en/developers/apps
         access_token = auth_model["auth_state"]["token_response"]["access_token"]
         token_type = auth_model["auth_state"]["token_response"]["token_type"]
-        granted_scopes = auth_model["auth_state"].get("scope", [])
+        granted_scopes = [
+            scope for scope in auth_model["auth_state"].get("scope", []) if scope
+        ]
         if not user_info["email"] and (
-            "user" in granted_scopes or "user:email" in granted_scopes
+            "user" in granted_scopes
+            or "user:email" in granted_scopes
+            or not granted_scopes
         ):
-            resp_json = await self.httpfetch(
+            resp = await self.httpfetch(
                 f"{self.github_api}/user/emails",
                 "fetching user emails",
                 method="GET",
+                parse_json=False,
+                raise_error=False,
```

Comment:
This is silently hiding all errors. Can you instead check for only the error case you want to catch, and otherwise allow the error to propagate upwards?

Reply:
Addressed in new commit